src/test/scala/com/databricks/labs/mosaic/datasource/GDALFileFormatTest.scala

```diff
@@ -0,0 +1,150 @@
+        assume(System.getProperty("os.name") == "Linux")
+
+        val grib = "/binary/grib-cams/"
+        val filePath = getClass.getResource(grib).getPath
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "NetCDF")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "NetCDF")
+            .load(filePath)
+            .select("proj4Str")
+            .take(1)
+
+    }
+
+    test("Read tif with GDALFileFormat") {
+        assume(System.getProperty("os.name") == "Linux")
+
+        val tif = "/modis/"
+        val filePath = getClass.getResource(tif).getPath
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "TIF")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "TIF")
+            .load(filePath)
+            .select("proj4Str")
+            .take(1)
+
+    }
+
+    test("Read zarr with GDALFileFormat") {
+        assume(System.getProperty("os.name") == "Linux")
+
+        val zarr = "/binary/zarr-example/"
+        val filePath = getClass.getResource(zarr).getPath
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("vsizip", "true")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "Zarr")
+            .option("vsizip", "true")
+            .load(filePath)
+            .take(1)
+
+        noException should be thrownBy spark.read
+            .format("gdal")
+            .option("driverName", "Zarr")
+            .option("vsizip", "true")
+            .load(filePath)
+            .select("proj4Str")
+            .take(1)
+
+    }
+
+    test("GDALFileFormat utility tests") {
+        val reader = new GDALFileFormat()
+        an[Error] should be thrownBy reader.prepareWrite(spark, null, null, null)
+
+        for (
+            driver <- Seq(
+                "GTiff",
+                "HDF4",
+                "HDF5",
+                "JP2ECW",
+                "JP2KAK",
+                "JP2MrSID",
+                "JP2OpenJPEG",
+                "NetCDF",
+                "PDF",
+                "PNG",
+                "VRT",
+                "XMP",
+                "COG",
+                "GRIB",
+                "Zarr"
+            )
+        ) {
+            GDALFileFormat.getFileExtension(driver) should not be ""
```

Comment:
Maybe better test that should not be "UNSUPPORTED"

Reply:
fair point, will update